src/main/cpp/blaze.cc

```diff
@@ -1433,7 +1433,7 @@ static map<string, EnvVarValue> PrepareEnvironmentForJvm() {
 
   const char *want_locale = "en_US.ISO-8859-1";
   bool override_locale = true;
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__HAIKU__)
```

Comment:
Is there a way to express this in a positive way? Something like
#if defined(LC_CTYPE_MASK) or defined(__USE_XOPEN2K8)

Reply:
Or maybe _XOPEN_VERSION >= 700

Reply:
`LC_CTYPE_MASK` sounds better to me, as it tests for the locale API directly rather than advertised conformace to the corresponding POSIX/XSI version (as to avoid false negatives).